EPPlus/ExcelAddress.cs

```
@@ -963,118 +963,35 @@ private void HandleBrackets(string first, string second, List<string> bracketPar
 		/// <summary>
 		/// Validates the given <see cref="Address"/>.
 		/// </summary>
-		/// <param name="Address">The address to validate.</param>
+		/// <param name="address">The address to validate.</param>
 		/// <returns>An <see cref="AddressType"/> indicating the address status.</returns>
-		internal static AddressType IsValid(string Address)
+		internal static AddressType IsValid(string address)
 		{
-			double d;
-			if (Address == "#REF!")
-			{
+			if (address == ExcelErrorValue.Values.Ref)
 				return AddressType.Invalid;
-			}
-			else if (double.TryParse(Address, NumberStyles.Any, CultureInfo.InvariantCulture, out d)) //A double, no valid address
-			{
+			else if (double.TryParse(address, NumberStyles.Any, CultureInfo.InvariantCulture, out var d)) // A double, not a valid address.
 				return AddressType.Invalid;
-			}
-			else if (IsFormula(Address))
-			{
+			else if (ExcelAddress.IsFormula(address))
 				return AddressType.Formula;
-			}
 			else
 			{
-				string wb, ws, intAddress;
-				if (SplitAddress(Address, out wb, out ws, out intAddress))
+				if (address.Contains(':'))
 				{
-					if (intAddress.Contains("[")) //Table reference
-					{
-						return string.IsNullOrEmpty(wb) ? AddressType.InternalAddress : AddressType.ExternalAddress;
-					}
-					else if (intAddress.Contains(","))
-					{
-						intAddress = intAddress.Substring(0, intAddress.IndexOf(','));
-					}
-					if (IsAddress(intAddress))
+					var rangeParts = address.Split(':');
+					AddressType? resultType = null;
+					foreach (var rangePart in rangeParts)

```

If there are more than two parts, should we throw an exception or return #REF or something?